cores/esp8266/core_esp8266_postmortem.cpp

```diff
@@ -280,22 +290,29 @@ static void postmortem_report(uint32_t sp_dump) {
     ets_printf_P(PSTR("<<<stack<<<\n"));
 
     // Use cap-X formatting to ensure the standard EspExceptionDecoder doesn't match the address
-    if (umm_last_fail_alloc_addr) {
+    if (_umm_last_fail_alloc.addr) {
 #if defined(DEBUG_ESP_OOM)
-        ets_printf_P(PSTR("\nlast failed alloc call: %08X(%d)@%S:%d\n"),
-            (uint32_t)umm_last_fail_alloc_addr, umm_last_fail_alloc_size,
-            umm_last_fail_alloc_file, umm_last_fail_alloc_line);
+        ets_printf_P(PSTR("\nlast failed alloc call: 0x%08X(%d), File: %S:%d\n"),
```

Comment:
`%S`, `umm_last_fail_alloc_file` can be  a `PROGMEM` string
At least some SDK file name strings are in `.irom.text`.

Reply:
No special case in newlib here? Implementation thinks of both 's' and 'S' as exactly same
https://github.com/earlephilhower/newlib-xtensa/blob/ebc967552ce827f21fc579fd8c437037c1b472ab/newlib/libc/stdio/nano-vfprintf_i.c#L217

[printf(3) ](https://man7.org/linux/man-pages/man3/printf.3.html)says it is '%ls', which is wide-char ('wchar\_t') and non-applicable here
> Synonym for ls.  Don't use.

Reply:
I have always been confused about this duality. Some print functions give an error (or is it a warning?) about %S and PROGMEM strings while other functions don't, like Serial.printf_P.

`%S` is used throughout `core_esp_8266_postmortem.cpp` for handling the printing of PROGMEM strings.

The comment for the code you pointed out reads to me that Arduino intends to have %S for PROGMEM strings.
```cpp
case 'S': // TODO: Verify cap-S under Arduino is "PROGMEM char*", not wchar_t
```

Reply:
6280e98b0360f85fdac2b8f10707fffb4f6e6e31 / #5376
requested in #4223 (...which I would guess gets dragged down from AVR Core?)
merged in the original newlib 3.x.x code via https://github.com/igrr/newlib-xtensa/pull/5

> Add "%S" (capital-S) format that I've been told, but cannot verify,                                                                                                                                                is used in Arduino to specify a PROGMEM string parameter in printfs,                                                                                                                                               as an alias for "%s" since plain "%s" can now handle PROGMEM. 

as-is both '%s' and even '%.*s' work seamlessly, no need for extra care
but it is true that it is used across the file, probably better served in a separate PR. at the very least for consistency